repair/repair.cc

```diff
@@ -1920,20 +1922,57 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
             auto& topology = erm->get_token_metadata().get_topology();
             std::unordered_map<dht::token_range, repair_neighbors> range_sources;
             auto nr_tables = get_nr_tables(db, keyspace_name);
-            rlogger.info("{}: started with keyspace={}, source_dc={}, nr_ranges={}, ignore_nodes={}", op, keyspace_name, source_dc, ranges.size() * nr_tables, ignore_nodes);
+            std::unordered_set<gms::inet_address> dc_nodes;
+            sstring ks_source_dc;
+            if (ignored_in_my_dc < 0) {
+                size_t ignored_in_my_dc = 0;
+                topology.for_each_node([&] (const locator::node* node) {
```

Comment:
We can do the dc_nodes calculation outside the loop.  The ignored_in_my_dc = -1 trick can be removed.

Reply:
true

Reply:
done in v2